instrumentation/github.com/gorilla/mux/otelmux/mux_test.go

```diff
@@ -162,3 +166,40 @@ func TestResponseWriterInterfaces(t *testing.T) {
 
 	router.ServeHTTP(w, r)
 }
+
+func TestCustomSpanNameFormatter(t *testing.T) {
+	exporter := tracetest.NewInMemoryExporter()
+	t.Cleanup(exporter.Reset)
+
+	tp := sdktrace.NewTracerProvider(sdktrace.WithSyncer(exporter))
+
+	routeTpl := "/user/{id}"
+
+	testdata := []struct {
+		spanNameFormatter func(string, *http.Request) string
+		expected          string
+	}{
+		{nil, routeTpl},
+		{func(string, *http.Request) string { return "custom" }, "custom"},
+		{func(name string, r *http.Request) string {
+			return fmt.Sprintf("%s %s", r.Method, name)
+		}, "GET " + routeTpl},
+	}
+
+	for _, d := range testdata {
```

Comment:
Could you use subtests here, to give each of them a name and allow running them individually?
https://go.dev/blog/subtests